src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -181,6 +188,31 @@ class CosmasGoogleCloudService(private val freeBucketName: String,
         responseObserver.onCompleted()
     }
 
+    private fun getFileIdVersionsInfo(info: ProjectInfo, fileId: String): List<FileVersionInfo> {
+        val infoName = "${info.projectId}-$fileId-fileIdVersionsInfo"
+        val infoBytes: Blob = this.storage.get(getBlobId(infoName, info))
+                ?: return listOf()
+        return FileIdVersionsInfo.parseFrom(infoBytes.getContent()).versionInfoList
+    }
+
+    private fun updateFileIdVersionsInfo(info: ProjectInfo, fileId: String, generation: Long?, timestamp: Long) {
+        val fileIdVersionsInfoList = getFileIdVersionsInfo(info, fileId)
+        val newFileVersionInfo = FileVersionInfo.newBuilder()
+                .setFileId(fileId)
+                // For in-memory storage implementation resultBlob.generation == null,
+                // but in this case we don't care about generation value, so I set it to 1L
+                .setGeneration(generation ?: 1L)
```

Comment:
Thanks for the comment. It probably makes sense to extract storage interface. We currently us a number of methods, such as "create regular object", "create cemetery", "create id mapping", "create version list", etc. If we extract an interface for them, we'll be able to store these different pieces of data in different places, if we need to. And yes, we'll be able to remove Elvis operator.

This is probably out of scope of this PR though, so feel free to just add TODO